ecs/context.go

```diff
@@ -55,45 +98,115 @@ func (h contextCreateAWSHelper) createProfile(name string) error {
+	if c.Profile == "default" {
+		c.Profile = ""
+	}
+	var description string
+
+	if c.CredsFromEnv {
+		if c.Description == "" {
+			description = "credentials read from environment"
+		}
+		return store.EcsContext{
+			CredentialsFromEnv: c.CredsFromEnv,
+			Profile:            c.Profile,
+			Region:             c.Region,
+		}, description
+	}
+
+	if c.Region != "" {
+		description = strings.TrimSpace(
+			fmt.Sprintf("%s (%s)", c.Description, c.Region))
 	}
-	description = strings.TrimSpace(
-		fmt.Sprintf("%s (%s)", description, region))
 	return store.EcsContext{
-		Profile: profile,
-		Region:  region,
+		Profile: c.Profile,
+		Region:  c.Region,
 	}, description
 }
 
-func (h contextCreateAWSHelper) createContextData(_ context.Context, opts ContextParams) (interface{}, string, error) {
-	profile := opts.Profile
-	region := opts.Region
-
-	profilesList, err := h.getProfiles()
+func (h contextCreateAWSHelper) selectFromLocalProfile(opts *ContextParams) error {
+	profilesList, err := getProfiles()
 	if err != nil {
-		return nil, "", err
+		return err
 	}
-	if profile != "" {
-		// validate profile
-		if profile != "default" && !contains(profilesList, profile) {
-			return nil, "", errors.Wrapf(errdefs.ErrNotFound, "profile %q", profile)
-		}
-	} else {
-		// choose profile
-		profile, err = h.chooseProfile(profilesList)
+	// choose profile
+	opts.Profile, err = h.chooseProfile(profilesList)
+	if err != nil {
+		return err
+	}
+
+	if opts.Region == "" {
+		region, isDefinedInProfile, err := getRegion(opts.Profile)
 		if err != nil {
-			return nil, "", err
+			return err
+		}
+		if isDefinedInProfile {
+			opts.Region = region
+		} else {
+			fmt.Println("No region defined in the profile. Choose the region to use.")
+			opts.Region, err = h.chooseRegion(opts.Region, opts.Profile)
+			if err != nil {
+				return err
+			}
 		}
 	}
-	if region == "" {
-		region, err = h.chooseRegion(region, profile)
-		if err != nil {
-			return nil, "", err
+	return nil
+}
+
+func (h contextCreateAWSHelper) createProfileFromCredentials(opts *ContextParams) error {
+	accessKey, secretKey, err := h.askCredentials()
+	if err != nil {
+		return err
+	}
+	opts.AccessKey = accessKey
+	opts.SecretKey = secretKey
+	// we need a region set -- either read it from profile or prompt user
+	// prompt for the region to use with this context
+	opts.Region, err = h.chooseRegion(opts.Region, opts.Profile)
+	if err != nil {
+		return err
+	}
+	// save as a profile
+	if opts.Profile == "" {
+		opts.Profile = opts.Name
+	}
+	fmt.Printf("Saving credentials under profile %s\n", opts.Profile)
+	return h.createProfile(opts.Profile, opts)
+}
+
+func (h contextCreateAWSHelper) createContextData(_ context.Context, opts ContextParams) (interface{}, string, error) {
+	if opts.CredsFromEnv {
+		ecsCtx, descr := h.createContext(&opts)
+		return ecsCtx, descr, nil
+	}
+	options := []string{
+		"Use AWS credentials from environment",
```

Comment:
Move to the bottom of the list

```suggestion
		"AWS environment variables",
```